services/storage-users/pkg/command/trash_bin.go

```diff
@@ -53,3 +89,402 @@ func PurgeExpiredResources(cfg *config.Config) *cli.Command {
+			case "skip":
+				overwriteOption = SKIP
+			case "replace":
+				overwriteOption = REPLACE
+			case "keep-both":
+				overwriteOption = KEEP_BOTH
+			default:
+				_ = cli.ShowSubcommandHelp(c)
+				return cli.Exit("The option flag is invalid", 1)
+			}
+			fmt.Printf("Restoring trash-bin items for spaceID: '%s' ...\n", spaceID)
+			ref, err := storagespace.ParseReference(spaceID)
+			if err != nil {
+				return err
+			}
+			client, err := pool.GetGatewayServiceClient(cfg.RevaGatewayGRPCAddr)
+			if err != nil {
+				log.Error().Err(err).Msg("error selecting next gateway client")
+				return err
+			}
+			ctx, err := utils.GetServiceUserContext(cfg.ServiceAccount.ServiceAccountID, client, cfg.ServiceAccount.ServiceAccountSecret)
+			if err != nil {
+				log.Error().Err(err).Msg("could not impersonate")
+				return err
+			}
+
+			spanOpts := []trace.SpanStartOption{
+				trace.WithSpanKind(trace.SpanKindClient),
+				trace.WithAttributes(
+					attribute.KeyValue{Key: "option", Value: attribute.StringValue(optionFlagVal)},
+					attribute.KeyValue{Key: "spaceID", Value: attribute.StringValue(spaceID)},
+				),
+			}
+			ctx, span := tp.Tracer("storage-users trash-bin restore-all").Start(ctx, "serve static asset", spanOpts...)
+			defer span.End()
+
+			res, err := client.ListRecycle(ctx, &provider.ListRecycleRequest{Ref: &ref, Key: "/"})
+			if err != nil {
+				log.Error().Err(err).Msg(retrievingErrorMsg)
+				return err
+			}
+			if res.Status.Code != rpc.Code_CODE_OK {
+				return fmt.Errorf("%s %s", retrievingErrorMsg, res.Status.Code)
+			}
+			if len(res.GetRecycleItems()) == 0 {
+				return cli.Exit("The trash-bin is empty. Nothing to restore", 0)
+			}
+
+			for {
+				fmt.Printf("Foud %d items that could be restored, continue (Y/n), show the items list (s): ", len(res.GetRecycleItems()))
+				var i string
+				_, err := fmt.Scanf("%s", &i)
+				if err != nil {
+					log.Err(err).Send()
+					continue
+				}
+				if strings.ToLower(i) == "y" {
+					break
+				} else if strings.ToLower(i) == "n" {
+					return nil
+				} else if strings.ToLower(i) == "s" {
+					for _, item := range res.GetRecycleItems() {
+						fmt.Printf("itemID: '%s', path: '%s', type: '%s', delited at: %s\n", item.GetKey(), item.GetRef().GetPath(), itemType(item.GetType()), utils.TSToTime(item.GetDeletionTime()).UTC().Format(time.RFC3339))
+					}
+				}
+			}
+
+			fmt.Printf("\nRun restoring-all with option=%s\n", optionFlagVal)
+			for _, item := range res.GetRecycleItems() {
+				fmt.Printf("restoring itemID: '%s', path: '%s', type: '%s'\n", item.GetKey(), item.GetRef().GetPath(), itemType(item.GetType()))
+				dstRes, err := restore(ctx, client, ref, item, overwriteOption)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("itemID: '%s', path: '%s', restored as '%s'\n", item.GetKey(), item.GetRef().GetPath(), dstRes.GetPath())
+			}
+			return nil
+		},
+	}
+}
+
+func restoreTrashBindItem(cfg *config.Config) *cli.Command {
```

Comment:
`restoreTrashBindItem` and `restoreAllTrashBinItems` are doing almost the same. Can't we refactor them to use common sub functions besides `restore`?